ruby/.rubocop.yml

```diff
@@ -0,0 +1,212 @@
+  # But it can be overridden by setting this parameter
+  IndentationWidth: ~
+
+Style/AndOr:
+  Description: Use &&/|| instead of and/or.
+  StyleGuide: https://github.com/bbatsov/ruby-style-guide#no-and-or-or
+  EnforcedStyle: conditionals
+  SupportedStyles:
+  - always
+  - conditionals
+
+Style/Documentation:
+  Description: Document classes and non-namespace modules.
+  Enabled: false
+  Exclude:
+    - 'spec/**/*'
+    - 'test/**/*'
+
+Style/DotPosition:
+  Description: Checks the position of the dot in multi-line method calls.
+  StyleGuide: https://github.com/bbatsov/ruby-style-guide#consistent-multi-line-chains
+  EnforcedStyle: trailing
+  SupportedStyles:
+  - leading
+  - trailing
+
+# Warn on empty else statements
+# empty - warn only on empty else
+# nil - warn on else with nil in it
+# both - warn on empty else and else with nil in it
+Style/EmptyElse:
+  EnforcedStyle: both
+  SupportedStyles:
+    - empty
+    - nil
+    - both
+
+Style/ExtraSpacing:
+  # When true, allows most uses of extra spacing if the intent is to align
+  # things with the previous or next line, not counting empty lines or comment
+  # lines.
+  AllowForAlignment: true
+  # When true, forces the alignment of = in assignments on consecutive lines.
+  ForceEqualSignAlignment: false
+
+Style/IfUnlessModifier:
+  Description: Favor modifier if/unless usage when you have a single-line body.
+  StyleGuide: https://github.com/bbatsov/ruby-style-guide#if-as-a-modifier
+  Enabled: true
+  MaxLineLength: 100
+
+# Checks the indentation of the first element in an array literal.
+Style/IndentArray:
+  # The value `special_inside_parentheses` means that array literals with
+  # brackets that have their opening bracket on the same line as a surrounding
+  # opening round parenthesis, shall have their first element indented relative
+  # to the first position inside the parenthesis.
+  #
+  # The value `consistent` means that the indentation of the first element shall
+  # always be relative to the first position of the line where the opening
+  # bracket is.
+  #
+  # The value `align_brackets` means that the indentation of the first element
+  # shall always be relative to the position of the opening bracket.
+  EnforcedStyle: special_inside_parentheses
+  SupportedStyles:
+    - special_inside_parentheses
+    - consistent
+    - align_brackets
+  # By default, the indentation width from Style/IndentationWidth is used
+  # But it can be overridden by setting this parameter
+  IndentationWidth: ~
+
+Style/MultilineOperationIndentation:
+  EnforcedStyle: aligned
+  SupportedStyles:
+    - aligned
+    - indented
+  # By default, the indentation width from Style/IndentationWidth is used
+  # But it can be overridden by setting this parameter
+  IndentationWidth: ~
+
+Style/SignalException:
+  Description: 'Checks for proper usage of fail and raise.'
+  StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#prefer-raise-over-fail'
+  EnforcedStyle: only_raise
```

Comment:
This is excellent and I'm very excited for this PR! Thank you again for
making this happen!

On Thu, Jun 30, 2016 at 10:56 AM, Moncef Belyamani <notifications@github.com
> wrote:

> In ruby/.rubocop.yml
> <https://github.com/18F/development-guide/pull/22#discussion_r69147951>:
>
> > +  # But it can be overridden by setting this parameter
> > +  IndentationWidth: ~
> > +
> > +Style/MultilineOperationIndentation:
> > +  EnforcedStyle: aligned
> > +  SupportedStyles:
> > +    - aligned
> > +    - indented
> > +  # By default, the indentation width from Style/IndentationWidth is used
> > +  # But it can be overridden by setting this parameter
> > +  IndentationWidth: ~
> > +
> > +Style/SignalException:
> > +  Description: 'Checks for proper usage of fail and raise.'
> > +  StyleGuide: 'https://github.com/bbatsov/ruby-style-guide#prefer-raise-over-fail'
> > +  EnforcedStyle: only_raise
>
> I don't have strong feelings either. I've just followed the Guide for
> this. They seem to have reversed their rule. Before, they preferred fail
> for exceptions, now they recommend raise. The semantic option is not
> documented, so I'm not sure when it would be appropriate to use fail.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/18F/development-guide/pull/22/files/b605d3b79f8e02563f915cf1ef7248090de12fe0#r69147951>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAH_Jr9iiK3457_nNFWBCnFmLRvTUeXks5qQ9kYgaJpZM4I-E10>
> .
>